### PR TITLE
add api for android read manifest xml file

### PR DESCRIPTION
**Pull request by magaofei**
Implement reading the Android manifest file api

### CODE REVIEW

File: MobSF/views/api/rest_api.py [hunk 1]
```diff
@@ -198,3 +201,16 @@ def api_view_source(request):
...
+
+
+@request_method(['POST'])
+@csrf_exempt
+def api_manifest(request):
+    """
+    """
+    params = ['bin', 'type', 'md5']
+    if set(request.POST) >= set(params):
+        resp = manifest_view.run(request, api=True)
```
Comment (ajinabraham) [inline MobSF/views/api/rest_api.py]: Handle request based on request type (ie. to support android, iOS and Windows)

File: MobSF/urls.py [hunk 2]
```diff
@@ -90,6 +90,7 @@
     url(r'^api/v1/download_pdf$', rest_api.api_pdf_report),
     url(r'^api/v1/report_json$', rest_api.api_json_report),
     url(r'^api/v1/view_source$', rest_api.api_view_source),
+    url(r'^api/v1/view_source/manifest_view$', rest_api.api_manifest),
```
[the remark this answers is not in the document]
Comment (magaofei) [reply MobSF/urls.py]: At first I tended to not use an extra API. Later I found that Android's Manifest file needs special processing to view it, so I used an extra API.

**Comment by magaofei** [comment]
support iOS Info.plist

File: MobSF/views/api/rest_api.py (hunk 1 above)
Comment (magaofei) [reply MobSF/views/api/rest_api.py]: I don't know about Windows App

File: MobSF/views/api/rest_api.py [hunk 3]
```diff
@@ -198,3 +201,16 @@ def api_view_source(request):
...
         response = make_api_response({"error": "Missing Parameters"}, 422)
     return response
+
+
+@request_method(['POST'])
+@csrf_exempt
+def api_manifest(request):
+    """
+    """
+    params = ['bin', 'type', 'md5']
```
[the remark this answers is not in the document]
Comment (ajinabraham) [reply MobSF/views/api/rest_api.py]: @magaofei This needs to be addressed.

**Comment by magaofei** [comment]
change md5 to hash
add Docs and tests

File: MobSF/views/api/rest_api.py [hunk 4]
```diff
@@ -198,3 +201,25 @@ def api_view_source(request):
...
+    """
+    :param type ipa/apk
+    """
+    params = ['bin', 'type', 'hash']
+    if set(request.POST) >= set(params):
+        scan_type = request.POST.get('type')
+        md5 = request.POST.get('hash')
+        if scan_type in ['eclipse', 'studio', 'apk']:
+            resp = manifest_view.run(request, api=True)
+        elif scan_type == 'ipa':
```
Comment (ajinabraham) [inline MobSF/views/api/rest_api.py]: How about `ios source code`.  Does this cover manifest file view for ios source code zip uploads?

File: StaticAnalyzer/views/ios/view_source.py [hunk 5]
```diff
@@ -46,6 +46,49 @@ def set_ext_api(file_path):
...
+                               md5 + '/Payload/')
+
+    info_plist_path = ''
+    dat = ''
+    for root, dirs, files in walklevel(src, 1):
+        for file in files:
+            if file == "Info.plist":
+                info_plist_path = os.path.join(root, "Info.plist")
+
+    print(info_plist_path)
```
Comment (ajinabraham) [inline StaticAnalyzer/views/ios/view_source.py]: Is this `print()` required?

File: StaticAnalyzer/views/ios/view_source.py [hunk 6]
```diff
@@ -46,6 +46,49 @@ def set_ext_api(file_path):
...
+def view_info_plist(md5):
+    """
+    """
+    src = os.path.join(settings.UPLD_DIR,
+                               md5 + '/Payload/')
+
+    info_plist_path = ''
+    dat = ''
+    for root, dirs, files in walklevel(src, 1):
+        for file in files:
```
Comment (ajinabraham) [inline StaticAnalyzer/views/ios/view_source.py]: `file` shouldn't be used for naming variables. It's a type clobbering
```
>>> file
<type 'file'>
``

File: templates/general/apidocs.html [hunk 7]
```diff
@@ -1,150 +1,153 @@
...
-<p><strong>Sample Call:</strong></p>
-<pre><code>curl -F 'file=@/Users/ajin/Desktop/diva-beta.apk' http://localhost:8000/api/v1/upload -H &quot;Authorization:563d64fc5054d3b239ac0419f1d6b2378465f5c80e1778c283eb1e3265bdd7ae&quot;
+
+    <ol>
+      <li><code>api/v1/upload</code> - <a href="#upload-file-api">Upload a File</a></li>
+      <li><code>api/v1/scan</code> - <a href="#scan-file-api">Scan a File</a></li>
+      <li><code>api/v1/delete_scan</code> - <a href="#delete-scan-api">Delete a Scan</a></li>
+      <li><code>api/v1/download_pdf</code> - <a href="#generate-pdf-report-api">Download PDF Report</a></li>
+      <li><code>api/v1/report_json</code> - <a href="#generate-json-report-api">Generate JSON Report</a></li>
+      <li><code>api/v1/view_source</code> - <a href="#view-source-api">View Source Files</a></li>
```
Comment (ajinabraham) [inline templates/general/apidocs.html]: Link for manifest view missing?

File: StaticAnalyzer/views/android/manifest_view.py [hunk 8]
```diff
@@ -37,6 +43,8 @@ def run(request):
                 'file': 'AndroidManifest.xml',
                 'dat': manifest
             }
+            if api:
+                return context
```
Comment (ajinabraham) [inline StaticAnalyzer/views/android/manifest_view.py]: for API, it should be 
```
return {'file': 'AndroidManifest.xml', 'contents': manifest }
```
Then again the similar response structure should be followed for IPA or APPX as well

File: StaticAnalyzer/views/ios/view_source.py [hunk 9]
```diff
@@ -46,6 +46,48 @@ def set_ext_api(file_path):
...
+    assert os.path.isdir(some_dir)
+    num_sep = some_dir.count(os.path.sep)
+    for root, dirs, files in os.walk(some_dir):
+        yield root, dirs, files
+        num_sep_this = root.count(os.path.sep)
+        if num_sep + level <= num_sep_this:
+            del dirs[:]
+
+
+def view_info_plist(md5):
```
Comment (ajinabraham) [inline StaticAnalyzer/views/ios/view_source.py]: Is this duplication? We already have a function to view Info Plist?

**Comment by ajinabraham** [comment]
Closing as the PR is not active.

**Pull request closed without merging**